### PR TITLE
Feature/130/create seller flow

### PR DESCRIPTION
**Pull request by GrazhdanochkaMariya**
- Added Basket to headers
- Seller can add, edit and remove his products

### CODE REVIEW

File: src/apps/inventory/urls.py [hunk 1]
```diff
@@ -18,4 +20,14 @@
     ),
     path("", ProductListView.as_view(), name="product_list"),
     path("seller/create_product/", ProductCreateView.as_view(), name="product_create"),
+    path(
+        "seller/edit/<int:pk>/<slug:category_slug>/<slug:product_slug>/",
```
Comment (gluhovSemen) [inline src/apps/inventory/urls.py]: why do we need <int:pk>, slug is already unic as far as i understand.

File: src/apps/inventory/urls.py [hunk 2]
```diff
@@ -18,4 +20,14 @@
...
+        ProductUpdateView.as_view(),
+        name="product_edit",
+    ),
+    path(
+        "seller/delete/<int:pk>/<slug:category_slug>/<slug:product_slug>/",
```
Comment (gluhovSemen) [inline src/apps/inventory/urls.py]: why do we need <int:pk>, slug is already unic as far as i understand.

File: src/apps/inventory/views.py [hunk 3]
```diff
@@ -96,3 +99,41 @@ def form_valid(self, form):
 
     def test_func(self):
         return self.request.user.user_type == "Seller"
+
+
+class ProductUpdateView(LoginRequiredMixin, UpdateView):
```
Comment (gluhovSemen) [inline src/apps/inventory/views.py]: maybe add UserPassesTestMixin i think it looks cooler, but thats my opinion

File: src/apps/inventory/views.py [hunk 4]
```diff
@@ -96,3 +99,41 @@ def form_valid(self, form):
...
+
+    def get_queryset(self):
+        category_slug = self.kwargs.get("category_slug")
+        product_slug = self.kwargs.get("product_slug")
+
+        queryset = Product.objects.filter(category__slug=category_slug, slug=product_slug)
+
+        return queryset
+
+    def is_seller(self):
```
Comment (gluhovSemen) [inline src/apps/inventory/views.py]: and replace this with test_func(self)

File: src/apps/inventory/views.py [hunk 5]
```diff
@@ -96,3 +99,41 @@ def form_valid(self, form):
...
+        return (
+            self.request.user.user_type == "Seller"
+            and self.request.user == self.get_object().seller
+        )
+
+
+class ProductDeleteView(LoginRequiredMixin, DeleteView):
```
Comment (gluhovSemen) [inline src/apps/inventory/views.py]: same about mixins

File: src/apps/user/templates/user/profile.html
Comment (gluhovSemen) [inline src/apps/user/templates/user/profile.html:1]: I am too bad at HTML so cant propaly review it, but for me looks nice))

File: src/apps/inventory/views.py (hunk 3 above)
Comment (egorvavilov) [reply src/apps/inventory/views.py]: great point, we need to test if our user is actual seller of product

File: src/apps/user/templates/user/profile.html
Comment (egorvavilov) [reply src/apps/user/templates/user/profile.html:1]: looks fine to me

File: src/apps/inventory/urls.py (hunk 1 above)
Comment (GrazhdanochkaMariya) [reply src/apps/inventory/urls.py]: True. It just didn`t work without it for me. I will fix it

Comment (egorvavilov) [reply src/apps/inventory/urls.py]: @GrazhdanochkaMariya take a look at get_object method

File: src/apps/user/templates/user/profile.html
Comment (GrazhdanochkaMariya) [reply src/apps/user/templates/user/profile.html:1]: > I am too bad at HTML so cant propaly review it, but for me looks nice))

Maybe UI looks nice, but I am not really good at HTML as well)

File: src/apps/inventory/views.py (hunk 4 above)
Comment (GrazhdanochkaMariya) [reply src/apps/inventory/views.py]: Have never used it before, Thanks!

**Review by egorvavilov — APPROVED** [review]

**Pull request merged**